addons/analytic/static/src/components/analytic_distribution/analytic_distribution.js

```diff
@@ -441,7 +456,10 @@ export class AnalyticDistribution extends Component {
         this.state.showDropdown = false;
     }
 
-    openAnalyticEditor() {
+    async openAnalyticEditor() {
+        if (this.needsPlans) {
+            await this.fetchPlansAndFormatData(this.props);
```

Comment:
Isn't it correct to call `willUpdate` instead?

Reply:
`willUpdate` checks `force_applicability` and does not fetch the plans
we need to do it at least once...when the widget is opened for the first time.

(I wonder if we need a loading indicator - or if there are no plans)